### PR TITLE
Add q17, 322, and 547 in Python

### PR DESCRIPTION
**Pull request by Hgjnnf**
Adding 17, 322, and 547 in Python. Test and test cases included

### CODE REVIEW

File: python/src/code_322.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+        def dp(num):
+            if num in memo:
+                return memo[num]
+            if num < 0:
+                return float("inf")
+            if num == 0:
+                return 0
+
+            res = float("inf")
```
Comment (dataprob) [inline python/src/code_322.py:16]: can do 1 of the following:
1. `from math import inf`
2. `import math`, `math.inf`

File: python/src/code_547.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from typing import List
+
+
+class Solution:
+    def findCircleNum(self, isConnected: List[List[int]]) -> int:
+        adj = {}
+        cities = [0] * len(isConnected)
```
Comment (dataprob) [inline python/src/code_547.py:7]: use a variable, like `N = len(isConnected)`, then use `N` in several places

File: python/src/code_17.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from typing import List
+
+
+class Solution:
+    lookup = {
```
Comment (dataprob) [inline python/src/code_17.py:5]: you used `self.lookup` later, so should it be `self.lookup` here?

File: python/src/code_322.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+from typing import List
+
+
+class Solution:
+    def coinChange(self, coins: List[int], amount: int) -> int:
+        memo = {}
+
+        def dp(num):
```
Comment (dataprob) [inline python/src/code_322.py:8]: type def in fns, `num: int`

**Review by dataprob — CHANGES_REQUESTED** [review]
Great work Gordon! See my comments

File: python/src/code_17.py (hunk 3 above)
Comment (Hgjnnf) [reply python/src/code_17.py:5]: No, self is apparently not defined at that scope. So  just lookup is fine.

**Review by dataprob — APPROVED** [review]
Great follow up Gordon, will merge!

**Pull request merged**